src/ORM/DatabaseAdmin.php

```diff
@@ -412,8 +406,23 @@ public function doBuild($quiet = false, $populate = true, $testMode = false)
      * @param string $fieldName    The field name to look in for obsolete class names
      * @param string $oldClassName The old class name
      * @param string $newClassName The new class name
+     * @deprecated This method performs inefficiently so it's been disabled
      */
     protected function updateLegacyClassNames($dataClass, $fieldName, $oldClassName, $newClassName)
+    {
+        // no-op
+        Deprecation::notice('5.0');
```

Comment:
Couldn't you leave the existing functionality intact by just adding

```suggestion
        Deprecation::notice('5.0');
        $this->updateLegacyClassNameField($dataClass, $fieldName, [$oldClassName => $newClassName]);
```

I think we should switch any core code to use the new API, but I don't really think we can just "disable" the functionality of a method in a patch version.

Reply:
True :D